lxd-agent/operations.go

```diff
@@ -52,11 +52,7 @@ func operationDelete(d *Daemon, r *http.Request) response.Response {
 		return response.SmartError(err)
 	}
 
-	err = op.Cancel()
-	if err != nil {
-		return response.BadRequest(err)
-	}
-
+	op.Cancel()
```

Comment:
We could check the status here and reject with bad request if already cancelled.
This would keep the API business logic in the hander.

What do you think?

Reply:
I wonder if we should reintroduce `MayCancel` and have handlers call that.

So `MayCancel` would be false in the API response if the operation is already cancelled or has completed. Then API handlers return a 400 if a request tries to cancel an operation where MayCancel is false